apps/system/utils/netcmd.c

```diff
@@ -113,19 +118,33 @@ static void nic_display_state(void)
 		} else {
 			struct ifreq tmp;
 			strncpy(tmp.ifr_name, ifr->ifr_name, IF_NAMESIZE);
-			ioctl(fd, SIOCGIFHWADDR, (unsigned long)&tmp);
+			ret = ioctl(fd, SIOCGIFHWADDR, (unsigned long)&tmp);
+			if (ret < 0) {
+				ndbg("fail %s:%d\n", __FUNCTION__, __LINE__);
+				continue;
+			}
 			sa = &tmp.ifr_hwaddr;
 			printf("Link encap: %s\t", ether_ntoa((struct ether_addr *)sa->sa_data));
 
-			ioctl(fd, SIOCGIFFLAGS, (unsigned long)ifr);
+			ret = ioctl(fd, SIOCGIFFLAGS, (unsigned long)ifr);
+			if (ret < 0) {
+				ndbg("fail %s:%d\n", __FUNCTION__, __LINE__);
+				continue;
+			}
 			printf("RUNNING: %s\n", (ifr->ifr_flags & IFF_UP) ? "UP" : "DOWN");
 		}
 		printf("\tinet addr: %s\t", inet_ntoa(sin->sin_addr));
 
-		ioctl(fd, SIOCGIFNETMASK, (unsigned long)ifr);
+		ret = ioctl(fd, SIOCGIFNETMASK, (unsigned long)ifr);
+		if (ret < 0) {
+			ndbg("fail %s:%d\n", __FUNCTION__, __LINE__);
+		}
 		sin = (struct sockaddr_in *)&ifr->ifr_addr;
 		printf("Mask: %s\t", inet_ntoa(sin->sin_addr));
-		ioctl(fd, SIOCGIFMTU, (unsigned long)ifr);
+		ret = ioctl(fd, SIOCGIFMTU, (unsigned long)ifr);
+		if (ret < 0) {
```

Comment:
I think when ioctl fails, this util should not print MTU set result.

Reply:
thanks, I'll fix it